src/asf_meta/meta2envi.c

```diff
@@ -108,12 +123,19 @@ envi_header* read_envi(char *envi_name)
     sprintf(envi->wavelength_units, "%s", value);
     }
     else if (strncmp(key, "band", 4)==0) {
-      fgets(line, 255, fp);
-      envi->band_name = trim_spaces(line);
-      int len = (int) strlen(envi->band_name);
-      if (envi->band_name[len-1] == '}' ||
-	  envi->band_name[len-1] == ',')
-	envi->band_name[len-1] = '\0';
+      if (strchr(line, '}') == NULL) {
+        fgets(line, 255, fp);
+        char *p = strchr(line, '}');
+        p[0] = '\0';
+        envi->band_name = trim_spaces(line);
+      }
+      else {
```

Comment:
Sadly enough you have you. PolSARPro header files had a carriage returns
that makes the next line belong to the same parameter. Example attached
(T11.bin.hdr). A user came up with an example from the Sentinel-1 Toolbox
(attached as well) that did not have those carriage return. That created
the segmentation fault to begin with.



Best regards,
Rudi

-- 
Rudi Gens
Alaska Satellite Facility, Geophysical Institute, University of Alaska
Fairbanks
903 Koyukuk Dr., P.O. Box 757320, Fairbanks, AK 99775-7320, USA
Phone: 1-907-4747621   Fax: 1-907-4746441
Email: rgens@alaska.edu   URL: http://www.gi.alaska.edu/~rgens

On Tue, Oct 27, 2015 at 4:26 PM, Kirk Hogenson <notifications@github.com>
wrote:

> In src/asf_meta/meta2envi.c
> <https://github.com/asfadmin/ASF_MapReady/pull/421#discussion_r43206685>:
>
> > @@ -108,12 +123,19 @@ envi_header* read_envi(char *envi_name)
> >      sprintf(envi->wavelength_units, "%s", value);
> >      }
> >      else if (strncmp(key, "band", 4)==0) {
> > -      fgets(line, 255, fp);
> > -      envi->band_name = trim_spaces(line);
> > -      int len = (int) strlen(envi->band_name);
> > -      if (envi->band_name[len-1] == '}' ||
> > -	  envi->band_name[len-1] == ',')
> > -	envi->band_name[len-1] = '\0';
> > +      if (strchr(line, '}') == NULL) {
> > +        fgets(line, 255, fp);
> > +        char *p = strchr(line, '}');
> > +        p[0] = '\0';
> > +        envi->band_name = trim_spaces(line);
> > +      }
> > +      else {
>
> In this branch of the if, you never read the line from the file
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asfadmin/ASF_MapReady/pull/421/files#r43206685>.
>